packages/test/test-end-to-end-tests/src/test/waitForSummary.spec.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { describeCompat } from "@fluid-private/test-version-utils";
+import type { IContainer } from "@fluidframework/container-definitions/internal";
+import type { ISharedMap } from "@fluidframework/map/internal";
+import {
+	type ITestObjectProvider,
+	type ITestContainerConfig,
+	createSummarizer,
+	summarizeNow,
+	type ChannelFactoryRegistry,
+	createAndAttachContainer,
+	DataObjectFactoryType,
+	type ITestFluidObject,
+} from "@fluidframework/test-utils/internal";
+
+const waitForSummary = async (
+	provider: ITestObjectProvider,
+	container: IContainer,
+	testContainerConfig: ITestContainerConfig,
+) => {
+	const { summarizer, container: summarizingContainer } = await createSummarizer(
+		provider,
+		container,
+		testContainerConfig,
+	);
+	await summarizeNow(summarizer);
+	summarizingContainer.close();
+};
+
+describeCompat("Wait for summary", "NoCompat", (getTestObjectProvider, apis) => {
+	const mapId = "map";
+	const { SharedMap } = apis.dds;
+	const registry: ChannelFactoryRegistry = [[mapId, SharedMap.getFactory()]];
+
+	it("Wait for summary", async function () {
+		const provider = getTestObjectProvider();
+		// This test fails on ODSP at least.
+		if (provider.driver.type !== "local") {
+			this.skip();
+		}
+		const testContainerConfig: ITestContainerConfig = {
+			fluidDataObjectType: DataObjectFactoryType.Test,
+			registry,
+		};
+
+		const loader = provider.makeTestLoader(testContainerConfig);
+		const container1 = await createAndAttachContainer(
+			provider.defaultCodeDetails,
+			loader,
+			provider.driver.createCreateNewRequest(provider.documentId),
+		);
+		provider.updateDocumentId(container1.resolvedUrl);
+		const dataStore1 = (await container1.getEntryPoint()) as ITestFluidObject;
+		const map1 = await dataStore1.getSharedObject<ISharedMap>(mapId);
+		map1.set("test op 1", "test op 1");
+		await waitForSummary(provider, container1, testContainerConfig);
+
+		map1.set("test op 2", "test op 2");
+		await waitForSummary(provider, container1, testContainerConfig);
```

Comment:
The description says that this fails with "disconnected" error. I believe it should also mention what the cause of failure was. Can you try turning verbose mode on and see what the summarization_cancel log says.